lib/parse_stream/svg.js

```diff
@@ -161,6 +161,7 @@ module.exports = function () {
 
           if (result) {
             parser.push(result);
+            state = STATE_IGNORE;
```

Comment:
I'd suggest move this one line up, to guarantee no races in previous push internals.

Reply:
Ok, but I am not sure what you mean, `push` does not call `transform` from itself again. According to documentation it just pushes its argument into stream's internal queue.

Reply:
Could you elaborate?

Reply:
If simplify everything, push() calls connected stream, which can do something causing push of next svg chunk. Probability if this case is zero.

Reply:
I did a little test. You are right that `push` on the first stream can lead to direct call (not from event loop) of `transform` of second stream into which is first stream piped. I misread documentation, it says that `push` will push data into the internal queue only when data is Buffer, Uint8Array or string (seemingly not when stream is not operating in object mode). But still your `parser` stream would need to be piped into itself to cause possible problems with `state`. And even then or in other circumstances `transform` function will not be called again until it calls `next` callback:

"transform._transform() is never called in parallel; streams implement a queue mechanism, and to receive the next chunk, callback must be called, either synchronously or asynchronously."